### PR TITLE
Implement intrinsic repeat

### PR DESCRIPTION
**Pull request by HarshitaKalani**
Fixes:  #2634

### CODE REVIEW

File: integration_tests/intrinsics_53.f90 [hunk 1]
```diff
@@ -3,6 +3,6 @@ program intrinsics_53
 integer(8) :: n8 = 3
 print *, repeat("a", n4)
 print *, repeat("a", n8)
-if (repeat("a", n4) /= "aaa") error stop
-if (repeat("a", n8) /= "aaa") error stop
+! if (repeat("a", n4) /= "aaa") error stop
```
Comment (HarshitaKalani) [inline integration_tests/intrinsics_53.f90:6]: This test is failing with the current implementation, however 
both repeat("a", n4) and repeat("a", n8) are equal to "aaa". I do not understand where does it fail.

**Comment by HarshitaKalani** [comment]
Also, the following test:
```fortran
program example
    implicit none
    character(len=5) :: s1
    character(len=10) :: s2
    s1 = "242"
    s2 = repeat(s1, 2)
    print *, '-', s2, '-'
end program example
```
gives -242242- as output. However, it according to GFortran is -242  242  - which is beacuse GFortran pads the string with spaces according to the len. This issue is mentioned here: https://github.com/lfortran/lfortran/issues/2864

File: src/libasr/pass/intrinsic_function_registry.h [hunk 2]
```diff
@@ -2888,6 +2895,124 @@ namespace Digits {
...
+    static inline ASR::expr_t* instantiate_Repeat(Allocator &al, const Location &loc,
+            SymbolTable *scope, Vec<ASR::ttype_t*>& arg_types, ASR::ttype_t *return_type,
+            Vec<ASR::call_arg_t>& new_args, int64_t /*overload_id*/) {
+        declare_basic_variables("_lcompilers_optimization_repeat_" + type_to_str_python(arg_types[0]));
+        fill_func_arg("x", arg_types[0]);
+        fill_func_arg("y", arg_types[1]);
+        auto result = declare(fn_name, arg_types[0], ReturnVar);
+
+        /*
+            function repeat(x, y) result(result)
```
Comment (Pranavchiku) [inline src/libasr/pass/intrinsic_function_registry.h:2965]: I suspect the type of variable `result` is incorrect and that is causing trouble, here is a repeat function written in fortran.

```fortran
function repeat_(s, n) result(r)
    character(len=*), intent(in) :: s
    integer, intent(in) :: n
    character(len=n*len(s)) :: r
    integer :: i 
    i = n
    if (i == 0) then
        r = ""
    else
        do while (i > 0)
            r = s // r
            i = i - 1
        end do
    end if
end function
```

Can you declare type of `result` variable correctly and check if it works?

Comment (HarshitaKalani) [reply src/libasr/pass/intrinsic_function_registry.h:2965]: The issue was with declaration of itr, resolved now. Thanks!

File: integration_tests/intrinsics_53.f90 (hunk 1 above)
Comment (HarshitaKalani) [reply integration_tests/intrinsics_53.f90:6]: Resolved!

**Review by certik — APPROVED** [review]
Looks great!

**Pull request merged**